### PR TITLE
Add --output option to dump json to file

### LINKED ISSUE #30: log levels
replace print with log levels (warning, debug, info)

### LINKED ISSUE #32: refactor t_global class
```
 12 class t_global(object):
 13     args = None

```
originally created to handle global vars from cmdline args

### LINKED ISSUE #35: add option to write output  to file (default is stdout)
include --output arg (default = bench-params.json)

### PR DESCRIPTION
**Pull request by rafaelfolco**
Default is stdout, --output is optional.
Replaced prints (stdout) to logging (stderr).
Replaced t_global with global var.
Code reorg with exception handling and functions.
--debug arg is added for enabling debug messages.

Closes #30
Closes #32
Closes #35

### CODE REVIEW

**Comment by k-rister** [comment]
I still hate the `enter` vs. `ctrl-enter` debate since i never know which to use...

**Comment by k-rister** [comment]
> Can we break this patch into 3 different commits that be reviewed individually?

1. first lets do the logger change
2. next lets do the global change
3. the --output change

File: multiplex.py [hunk 1]
```diff
@@ -156,55 +168,73 @@ def load_requirements(req_arg):
...
     return(req_json)
 
-
-def main():
-    input_json = None
-
+def load_input_file(mv_file):
     try:
-        input_fp = open(t_global.args.input, 'r')
+        input_fp = open(mv_file, 'r')
```
Comment (k-rister) [inline multiplex.py:178]: does this belong in this one?

File: multiplex.py [hunk 2]
```diff
@@ -5,8 +5,15 @@
...
 import os
+import logging
 
 from jsonschema import validate
+from jsonschema import exceptions
+
+EC_SUCCESS=0
+EC_SCHEMA_FAIL=1
+EC_JSON_FAIL=2
+EC_REQUIREMENTS_FAIL=3
```
Comment (k-rister) [inline multiplex.py:16]: so I'm ignorant...what does EC mean?

Comment (rafaelfolco) [reply multiplex.py:16]: > so I'm ignorant...what does EC mean?

Exit Code --> enumerate exit codes for consistency

File: multiplex.py (hunk 1 above)
Comment (rafaelfolco) [reply multiplex.py:178]: no, I'll fix it

File: multiplex.py [hunk 3]
```diff
@@ -156,55 +168,73 @@ def load_requirements(req_arg):
...
+    except IOError:
+        log.exception("Could not open schema file %s" % (json_schema_file))
+        return(False)
+    except ValueError:
+        log.exception("Could not load a valid JSON schema from %s"
+                      % (json_schema_file))
+        return(False)
+    except ValueError:
+        log.exception("JSON validation failed for %s using schema %s"
+                      % (t_global.args.input, json_schema_file))
```
Comment (rafaelfolco) [inline multiplex.py]: this is wrong, will fix it

**Comment by rafaelfolco** [comment]
> Can we break this patch into 3 different commits that be reviewed individually?

done

**Review by k-rister — APPROVED** [review]

**Pull request merged**